src/utils/AccessibleModal.ts

```diff
@@ -0,0 +1,90 @@
+import { ModalBox as ModalBoxModule } from '../ExternalModulesShim';
+import { FocusTrap } from '../ui/FocusTrap/FocusTrap';
+import { l } from '../strings/Strings';
+import { $$ } from './Dom';
+import { KeyboardUtils, KEYBOARD } from './KeyboardUtils';
+
+export interface IAccessibleModalOptions {
+  overlayClose?: boolean;
+  sizeMod: 'big';
+}
+
+export class AccessibleModal {
+  private focusTrap: FocusTrap;
+  private activeModal: Coveo.ModalBox.ModalBox;
+  private options: IAccessibleModalOptions;
+
+  public get isOpen() {
+    return !!this.focusTrap;
+  }
+
+  public get element() {
+    return this.activeModal && this.activeModal.modalBox;
+  }
+
+  public get content() {
+    return this.activeModal && this.activeModal.content;
+  }
+
+  public get wrapper() {
+    return this.activeModal && this.activeModal.wrapper;
+  }
+
+  constructor(
+    private className: string,
+    private ownerBody: HTMLBodyElement,
+    private modalboxModule: Coveo.ModalBox.ModalBox = ModalBoxModule,
+    options: Partial<IAccessibleModalOptions> = {}
+  ) {
+    this.options = {
+      ...{
+        sizeMod: 'big'
+      },
+      ...options
+    };
+  }
+
+  public open(title: HTMLElement, content: HTMLElement, validation: () => boolean) {
+    if (this.isOpen) {
+      return;
+    }
+    this.activeModal = this.modalboxModule.open(content, {
+      title,
+      className: this.className,
+      validation: () => {
+        this.onModalClose();
+        return validation();
+      },
+      body: this.ownerBody,
+      sizeMod: this.options.sizeMod,
+      overlayClose: this.options.overlayClose
+    });
+    this.focusTrap = new FocusTrap(this.element);
```

Comment:
I'm curious as to what `focusTrap` is?

Reply:
`focusTrap` prevents the keyboard focus from leaving a component. It also prevents accessibility focus from VoiceOver or TalkBack from leaving a component.